File: tmp/security/authorization/src/main/java/org/tsdes/intro/spring/security/authorization/Application.java
package org.tsdes.intro.spring.security.authorization;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

/**
 * Created by arcuri82 on 12-Dec-17.
 */
@SpringBootApplication
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }


    @Bean
    public PasswordEncoder passwordEncoder(){
        return new BCryptPasswordEncoder();
    }
}

File: advanced/security/database/src/main/kotlin/org/tsdes/spring/security/database/Application.kt
package org.tsdes.spring.security.database

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

/**
 * Created by arcuri82 on 08-Nov-17.
 */
@SpringBootApplication
@ComponentScan
class Application{

    /**
     * Make sure we use BCrypt to encode/hash the password.
     * To override the default encoding, we need to provide
     * a bean with name passwordEncoder
     */
    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }

}


fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}

